src/Tinker.jl

```diff
@@ -498,12 +496,49 @@ push!(zoom_ctrl["enabled"], false)
         drawrecthandle(ctx, rh, colorant"blue", colorant"white")
     end
 
-showall(win);
+    showall(win);
 
-append!(c.preserved, [zoom_ctrl, pandrag, rectselect])
+    append!(c.preserved, [zoom_ctrl, pandrag, rectselect])
 
+    imagectx = Dict("Image"=>image, "Canvas"=>c, "ZoomRegion"=>zr,"Rectangle"=>
+                    rect, "MouseActions"=>[pandrag, zoom_ctrl, rectselect])
+    push!(img_ctxs, push!(value(img_ctxs), imagectx))
+    return imagectx
 end;
 
 init_gui(file::AbstractString) = init_gui(load(file); name=file)
 
+active_context = map(img_ctxs) do ic # signal dependent on img_ctxs
+    if isempty(ic)
+        # placeholder value of appropriate type
+        init_gui(testimage("lighthouse")) # find a better placeholder
+    else
+        ic[end] # currently gets last element of img_ctxs
+    end
+end
+
+# Signals for active_context
+#=
+xzoom = map(value(active_context)["ZoomRegion"]) do r # holds x zoom level
+    r.fullview.x.right/(r.currentview.x.right-(r.currentview.x.left-1))
+end
+
+yzoom = map(value(active_context)["ZoomRegion"]) do r # hold y zoom level
+    r.fullview.y.right/(r.currentview.y.right-(r.currentview.y.left-1))
+end
+=#
+function set_mode(mode::Int)
```

Comment:
I would just pass `active_context` as an argument to this function. Especially with Signals updating asynchronously there could be weird things going on, e.g., what if something in this function [`yield`s](https://docs.julialang.org/en/latest/stdlib/parallel/#Base.yield) (`println`, below, is an example of such a function) and during the yield some other Reactive signal swaps the global `active_context` out on you? You'll get a really rare bug where half of this function executed for one context and half for a different context.

Asynchronous programming is weird! But very useful in GUIs.